napari/_qt/widgets/qt_dims_slider.py

```diff
@@ -598,7 +603,18 @@ def __init__(self, slider):
         self.dims.events.current_step.connect(self._on_axis_changed)
         self.current = max(self.dims.current_step[self.axis], self.min_point)
         self.current = min(self.current, self.max_point)
-        self.timer = QTimer()
+
+        self.timer.setSingleShot(True)
```

Comment:
I think a single-shot `QTimer` will only fire once no matter how many time you call `start`:

https://doc.qt.io/qt-5/qtimer.html#singleShot-prop
https://doc.qt.io/qt-5/qtimer.html#start

After pulling down this branch, playback no longer works - it just advances once.

So either this should not be a single-shot timer, or we should go back to the static call to `QTimer::singleShot`.

Reply:
Are we still able to use `QTimer::singleShot` with the qtimer patching in test?

Reply:
> Are we still able to use `QTimer::singleShot` with the qtimer patching in test?

Then we need to find another way to detect that there is no pending timer before test end, 

But this looks like a threading problem. In my console, I have:

```
WARNING: QObject::startTimer: Timers cannot be started from another thread
23:18:59 WARNING QObject::startTimer: Timers cannot be started from another thread
```

Reply:
Yeah, I have that in my console too. I suspect `QTimer::singleShot` bypasses that by creating a single-shot `QTimer` on the main thread, or just enqueuing the appropriate event?

Reply:
Now it working

Reply:
Retested and it's working now. So I guess you can start a single-shot timer more than once - good to know!